src/test/java/cucumber/StepDefs.java

```diff
@@ -226,6 +226,10 @@ private int getMessageCountBasedOnProblemType(String problemEnum, JsonObject rep
                 count = this.getMessageCount(reportJson, ProblemType.ARRAY_VALUE_OUT_OF_MIN_MAX_RANGE.getKey());
             } else if (strTemp.equals("NO_PRODUCTS_FOUND")) {
                 count = this.getMessageCount(reportJson, ProblemType.NO_PRODUCTS_FOUND.getKey());
+            } else if (strTemp.equals("TABLE_FILE_READ_ERROR")) {
+                count = this.getMessageCount(reportJson, ProblemType.TABLE_FILE_READ_ERROR.getKey());
+            } else if (strTemp.equals("FIELD_VALUE_OUT_OF_MIN_MAX_RANGE")) {
+                count = this.getMessageCount(reportJson, ProblemType.FIELD_VALUE_OUT_OF_MIN_MAX_RANGE.getKey());
```

Comment:
This `getMessageCountBasedOnProblemType`function omg my eyes 👀 😆

Maybe use introspection here? The reflection API could turn this 113 line function into 8–10 lines 🤔

Reply:
Yeah... It pains me to add stuff to it, but I just ran out of time. I will create a ticket to refactor this.